middleware/dedupe.js

```diff
@@ -0,0 +1,95 @@
+var logger = require('pelias-logger').get('api');
+var _ = require('lodash');
+
+function setup() {
+  return dedupeResults;
+}
+
+function dedupeResults(req, res, next) {
+  // do nothing if no result data set
+  if (_.isUndefined(req.clean) || _.isUndefined(res) || _.isUndefined(res.data)) {
+    return next();
+  }
+
+  // loop through data items and only copy unique items to uniqueResults
+  var uniqueResults = [];
+
+  _.some(res.data, function (hit) {
```

Comment:
you can use the native js versions instead of the `lodash` ones if you like, they were introduced in ECMA 5.1 and some old browsers haven't caught up *cough internut exploder* yet

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/some
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/every